oshi-core/src/main/java/oshi/software/common/AbstractOSThread.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2010 - 2020 The OSHI Project Contributors: https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.software.common;
+
+import java.util.Collections;
+import java.util.List;
+
+import oshi.software.os.OSProcess;
+import oshi.software.os.OSThread;
+
+public abstract class AbstractOSThread implements OSThread {
+
+    private final OSProcess parentProcess;
+
+    public AbstractOSThread(OSProcess process) {
+        this.parentProcess = process;
+    }
+
+    @Override
+    public OSProcess getParentProcess() {
+        return this.parentProcess;
+    }
+
+    @Override
+    public int getProcessID() {
+        return getParentProcess().getProcessID();
+    }
+
+    @Override
+    public int getParentProcessID() {
+        return getParentProcess().getParentProcessID();
+    }
+
+    @Override
+    public List<OSThread> getThreadDetails() {
+        return Collections.emptyList();
```

Comment:
Hmm, why are we returning ThreadDetails from the OSThread?   I guess we want to override the parent that returns the list of threads?  Not sure we need this overridden method, or if it should throw an exception.

Reply:
My thought was to return empty list as threads won't have anything to return. I am not quite sure about throwing an exception for accessing a method.

Reply:
It's reasonable to throw an `UnsupportedOperationException` for a method that shouldn't be called.  But see my other comments about perhaps not extending OSProcess overall.  This, combined with the need for different "usage" method naming, is a strong argument for a standalone class that we can name things like we want and not be forced to inherit/implement things that provide no added value.